src/types.ts

```diff
@@ -1,29 +1,45 @@
-
-
-export type SectionData = {
-    name: string,
-    description: string,
-    numRevisions: number
+export type ProjectData = {
+    id: string;
 }
 
-
-
-export type ChordProgression = string[]
-
-
-export type File = {
-    title: string,
-    numComments: number,
-    id: string,
+export enum EntityType {
+    PROJECT = 'project',
+    SECTION = 'section',
+    FILE = 'file',
 }
 
+export type EntityPointer = {
+    entityType: EntityType;
+    entityId: string;
+}
```

Comment:
Here we define the `EntityPointer` type, which is used to attach objects to other objects